amespahdbpythonsuite/xmlparser.py

```
@@ -0,0 +1,525 @@
+    """Parse a PAHdb XML file.
+
+    Optional behaviour includes validating the XML schema.
+
+    Attributes:
+        filename (str): XML file path.
+        validate (bool): Whether to validate the XML schema or not.
+        schema_is_valid (bool): Outcome of the schema checking.
+        database (dict): Dictionary containing the PAHdb-parsed data.
+        info (dict): Dictionary containing the PAHdb-parsed metadata.
+
+    Note:
+        `validated` (bool) indicates if the schema is valid (True) or
+            not (False). Set to None until validation is performed.
+
+    Examples:
+        parser = XMLparser(filename="xml_file.xml")
+        parser.verify_schema()
+        database, info = parser.to_pahdb_dict()
+
+        parser = XMLparser(filename="xml_file.xml", validate=True)
+        database, info = parser.to_pahdb_dict()
+
+        parser = XMLparser()
+        parser.filename = "xml_file.xml"
+        parser.validate = True
+        database, info = parser.to_pahdb_dict()
+
+        parser = XMLparser.from_string("<XML code here>")
+        parser.verify_schema()
+        database, info = parser.to_pahdb_dict()
+
+    """
+
+    def __init__(self, filename=None, validate=False):
+        """Inits XMLparser with schema checking off, no given filename."""
+        self.filename = filename
+        self.validate = validate
+        self.schema_is_valid = None
+
+        self.database = None
+        self.info = None
+
+    def __repr__(self):
+        """Class representation."""
+        return (f'{self.__class__.__name__}('
+                f'filename={self.filename!r}, '
+                f'validate={self.validate!r})')
+
+    def __str__(self):
+        """A description of the object."""
+        return (f'An XML parser from the AmesPAHdbPythonSuite. '
+                f'XML file: \n{self.filename!r}.')
+
+    def verify_schema(self):
+        """Validate XML schema.
+
+        Note:
+            Requires that self.filename is set.
```

The method should throw an exception when self.filename is not set and/or when no linked schema is found.